python/cudf/cudf/_lib/column.pyx

```diff
@@ -155,7 +155,11 @@ cdef class Column:
             )
 
         if value is not None:
-            required_size = bitmask_allocation_size_bytes(self.base_size)
+            # base_size may be zero for a struct column with no
+            # children, but the column may have non-zero length.
+            required_size = bitmask_allocation_size_bytes(
+                max(self.base_size, self._size)
+            )
```

Comment:
As below, I think both of these options are wrong. I think that the required size for the bitmask must be _at least_ `bitmask_allocation_size_bytes(self._size + self._offset)`.

So, suggestion:

```suggestion
            # bitmask size must be relative to offset = 0 data.
            required_size = bitmask_allocation_size_bytes(self._size + self._offset)
```